# HG changeset patch
# User Magnus Melin <mkmelin+mozilla@iki.fi>
# Date 1699608929 -7200
# Node ID e4b2a40488f4c71ec223eeb925723681d5edcda6
# Parent  0923a3d30707e72cfc266c32ef497774eb0ff154
Bug 1864124 - Handle advanced address book search with empty name. r=john.bieling

Don't throw, show the address book listing without limitation instead.

Differential Revision: https://phabricator.services.mozilla.com/D193275

diff --git a/mail/components/addrbook/content/abSearchDialog.js b/mail/components/addrbook/content/abSearchDialog.js
--- a/mail/components/addrbook/content/abSearchDialog.js
+++ b/mail/components/addrbook/content/abSearchDialog.js
@@ -185,16 +185,19 @@ function onSearch() {
   gSearchSession.searchTerms = saveSearchTerms(
     gSearchSession.searchTerms,
     gSearchSession
   );
 
   let searchUri = "?(";
   for (let i = 0; i < gSearchSession.searchTerms.length; i++) {
     let searchTerm = gSearchSession.searchTerms[i];
+    if (!searchTerm.value.str) {
+      continue;
+    }
     // get the "and" / "or" value from the first term
     if (i == 0) {
       if (searchTerm.booleanAnd) {
         searchUri += "and";
       } else {
         searchUri += "or";
       }
     }
@@ -325,16 +328,20 @@ function onSearch() {
         opStr +
         "," +
         encodeABTermValue(searchTerm.value.str) +
         ")";
     }
   }
 
   searchUri += ")";
+  if (searchUri == "?()") {
+    // Empty search.
+    searchUri = "";
+  }
   SetAbView(currentAbURI, searchUri, "");
 }
 
 // used to toggle functionality for Search/Stop button.
 function onSearchButton(event) {
   if (
     event.target.label ==
     gSearchBundle.GetStringFromName("labelForSearchButton")

